scrapy_zyte_api/responses.py

```diff
@@ -113,7 +118,9 @@ def _prepare_headers(cls, api_response: Dict[str, Any]):
 
 class ZyteAPITextResponse(ZyteAPIMixin, HtmlResponse):
     @classmethod
-    def from_api_response(cls, api_response: Dict, *, request: Request = None):
+    def from_api_response(
+            cls, api_response: Dict, maxsize: Optional[int], warnsize: Optional[int], *, request: Request = None
```

Comment:
maxsize and warnsize look out of place here. It could make sense to have them here if some work can be prevented, but with the current implementation it seems you can just as easy check len(response.body) after the response is created. 

Because of this, it looks like we can simplify the code by moving the response.body size check to `_download_request` (not even to `_process_response`). This way everything related to max size (reading the settings, showing warnings, checking the body size) would be in the middleware, there is no need to pass parameters around.